pkg/store/proxy_heap.go

```diff
@@ -170,9 +170,13 @@ func (h *ProxyResponseHeap) Less(i, j int) bool {
 	jResp := (*h)[j].rs.At()
 
 	if iResp.GetSeries() != nil && jResp.GetSeries() != nil {
+		// Response sets are sorted before adding external labels.
+		// This comparison excludes those labels to keep the same order.
+		iStoreLbls := (*h)[i].rs.StoreLabels()
+		jStoreLbls := (*h)[j].rs.StoreLabels()
 		iLbls := labelpb.ZLabelsToPromLabels(iResp.GetSeries().Labels)
 		jLbls := labelpb.ZLabelsToPromLabels(jResp.GetSeries().Labels)
-		return labels.Compare(iLbls, jLbls) < 0
+		return labels.Compare(rmLabels(iLbls.Copy(), iStoreLbls), rmLabels(jLbls.Copy(), jStoreLbls)) < 0
```

Comment:
I think we need to compare the full label set when the comparison without external labels is 0.

Reply:
You are right. One example I see is when the responses have repeated series, but their stores had different external labels. it has to compare the full set to keep the repeated series together.
I have added a test with this case.